### ISSUE TITLE
SEO Title, Description and Site Name all wrong on multi-site setup

### ISSUE THREAD

**Issue by Coalescent** [barrelstrength/craft-sprout-seo#193]
### Description (screenshots, logs, ...)

I'm seeing several issues:

1. A Secondary site SEO Title does not include the entry title for a given page (just using the seo optimize tag) - though the entry title _does_ show up on the primary website (usually). See number 2.
1. Custom descriptions do not seem to be output when adding them to a Sprout SEO field for a given entry (when the SEO field is set to ‘Display editable field’ for the description). Further though, when adding a custom description to the SEO field and saving the entry, it seems to make the SEO title output incorrect). See the attached image which will illustrate this better.
1. I am seeing a Site Name string output in the SEO Title that no longer exists in the install. It _did_ exist when this was a Craft 2 website some time ago (but that site name was changed some time ago and the Website Identity does not contain such a string). See the attached image which will illustrate this better.

![Sprout SEO Issue](https://user-images.githubusercontent.com/6394091/74988476-ca554b80-53f2-11ea-8c78-70916326eb1a.png)

### Steps to reproduce

1. Add a custom description as per the screen above and check the output
2. As to the incorrect (old) Site Name, I have no idea why that's happening or how to reproduce (though oddly, I believe that _was not_ happening prior to a Craft 3.4 update (but I could be wrong)). **Update:** If I output `{{siteName}}` in any templates, I get the expected string - not the old one showing up via the seo optimize's generated title string.

### Additional info

PHP	7.3.8
Database driver & version	MySQL 5.7.26
Craft edition & version	Craft Pro 3.4.6.1
Sprout SEO	4.3.3

**Comment by BenParizek** [comment]
Thanks @Coalescent 

We're working on a fix to a few issues right now. I'll be sure to review these items and get them addressed in the next release. I'll update this thread when we do.

**Comment by BenParizek** [comment]
@Coalescent We've just released v4.4.1 that should address this and several other bugs. Let me know if you see anything else out of place.

**Comment by Coalescent** [comment]
Hmmm... migration failed...

```
One of Sprout SEO’s migrations failed.

yii\base\ErrorException: Array to string conversion

Migration: barrelstrength\sproutseo\migrations\m200224_000004_update_globals_identity_image_setting

Output:

> update in {{%sproutseo_globals}} ... done (time: 0.002s)
> update in {{%sproutseo_globals}} ...Exception: Array to string conversion (/Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/ColumnSchema.php:157)
#0 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(8, 'Array to string...', '/Users/Mike/Sit...', 157)
#1 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/ColumnSchema.php(157): craft\web\ErrorHandler->handleError(8, 'Array to string...', '/Users/Mike/Sit...', 157, Array)
#2 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/mysql/ColumnSchema.php(30): yii\db\ColumnSchema->typecast(Array)
#3 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/mysql/ColumnSchema.php(49): craft\db\mysql\ColumnSchema->typecast(Array)
#4 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/QueryBuilder.php(647): yii\db\mysql\ColumnSchema->dbTypecast(Array)
#5 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/QueryBuilder.php(625): yii\db\QueryBuilder->prepareUpdateSets('{{%sproutseo_gl...', Array, Array)
#6 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/db/Command.php(580): yii\db\QueryBuilder->update('{{%sproutseo_gl...', Array, Array, Array)
#7 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/Command.php(141): yii\db\Command->update('{{%sproutseo_gl...', Array, Array, Array)
#8 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/Migration.php(276): craft\db\Command->update('{{%sproutseo_gl...', Array, Array, Array, false)
#9 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/barrelstrength/sprout-seo/src/migrations/m200224_000004_update_globals_identity_image_setting.php(46): craft\db\Migration->update('{{%sproutseo_gl...', Array, Array, Array, false)
#10 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/Migration.php(52): barrelstrength\sproutseo\migrations\m200224_000004_update_globals_identity_image_setting->safeUp()
#11 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/MigrationManager.php(233): craft\db\Migration->up(true)
#12 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/db/MigrationManager.php(153): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutseo\migrations\m200224_000004_update_globals_identity_image_setting))
#13 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/services/Updates.php(233): craft\db\MigrationManager->up()
#14 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(510): craft\services\Updates->runMigrations(Array)
#15 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/controllers/UpdaterController.php(203): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#16 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#17 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#18 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#19 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('migrate', Array)
#20 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('migrate', Array)
#21 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('updater/migrate', Array)
#22 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/web/Application.php(646): craft\web\Application->runAction('updater/migrate')
#23 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/craftcms/cms/src/web/Application.php(243): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#24 /Users/Mike/Sites/cadexcraft3/cadex/craft/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#25 /Users/Mike/Sites/cadexcraft3/cadex/cadex.com/html/index.php(21): yii\base\Application->run()
#26 {main}
```

**Comment by BenParizek** [comment]
@Coalescent I've just released a fix (v4.4.1.1). We were overlooking a scenario where you may not have an image set in your Global metadata.

**Comment by Coalescent** [comment]
Appreciate the quick fix @BenParizek but I'm finding some new issues now. I'm getting errors on the front and back end for certain entries. See screens below... some entries work and some don't (for the exact same section/structure and entry type).

CP Error when accessing entry edit page:

![Screen Shot 2020-03-03 at 12 16 09 PM](https://user-images.githubusercontent.com/6394091/75816018-3ddb4f00-5d49-11ea-9369-e1d3cb7d98e8.png)

Front end error:

![Screen Shot 2020-03-03 at 12 16 18 PM](https://user-images.githubusercontent.com/6394091/75816054-4e8bc500-5d49-11ea-8b17-092e4a394d8e.png)

**Comment by BenParizek** [comment]
@Coalescent Would you be able to provide me a screenshot of your Element Metadata field settings for one of the entries with this issue?

Also, could you confirm this is for an existing entry or a new entry?

**Comment by Coalescent** [comment]
Yes, these are for existing entries. See screen below. Thanks.

![Screen Shot 2020-03-03 at 1 00 43 PM](https://user-images.githubusercontent.com/6394091/75819229-1d15f800-5d4f-11ea-9264-4f3344420cb1.png)

**Comment by BenParizek** [comment]
@Coalescent I've just published v4.4.1.2 which should defend against the scenarios your running into a bit better. I'm not entirely sure what conditions need to be present for those errors to arise yet, so let me know if you continue to run into any issues.

**Comment by Coalescent** [comment]
Thanks @BenParizek. That appears to have fixed those issues in question... I'll put these updates through some closer testing later today/tomorrow.
